### PR TITLE
[WIP] Add MDN-style documentation for the new proposal

### PR DESCRIPTION
**Pull request by AlexLakatos**

### CODE REVIEW

File: docs/monetization-event-asset-code.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+import Specifications from '@site/src/components/Specifications';
+import BrowserCompat from '@site/src/components/BrowserCompat';
+
+The **`assetCode`** property of the [MonetizationEvent](monetization-event.md) interface returns the currency code.
+
+## Value
+
+A string representing the three-character currency code according to the last receipt.
```
Comment (sabineschaller) [inline docs/monetization-event-asset-code.md:14]: Again, if we want to deprecate receipts, I wouldn't want to mention them that often.

File: docs/web-monetization-api.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+---
+id: web-monetization-api
+title: Web Monetization API
+sidebar_label: Web Monetization API
+---
+
+import Specifications from '@site/src/components/Specifications';
+import BrowserCompat from '@site/src/components/BrowserCompat';
+
+The Web Monetization API provides a seamless payment experience for site owners and visitors. Unlike the W3C's Payment Request API, when a web-monetized visitor visits a web-monetized website, a continuous stream of micro-payments takes place automatically. Thus, visitors automatically send a small sum per time unit to any web-monetized website they visit. The payment rate is predetermined by the user's web monetization agent and adapts over time.
```
[the remark this answers is not in the document]
Comment (AlexLakatos) [reply docs/web-monetization-api.md:10]: Open Payments doesn't preclude it from being continuous. Those aren't mutually exclusive.

File: docs/web-monetization-api.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## Web Monetization Concepts and Usage
+
+The Web Monetization API allows web apps to accept payment streams from users. There are two components to this API:
+
+- The Web Monetization Agent is designed to be implemented in browsers and expose the Web Monetization API. It's responsible to:
+  - get the rate, token, and BTP address from the Web Monetization Provider
```
[the remark this answers is not in the document]
Comment (AlexLakatos) [reply docs/web-monetization-api.md:17]: Definitely! We'll need to update it after we have an OP implementation for WM.

File: docs/web-monetization-api.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  - get the payment information from the `<link>` element
+  - create an [Interledger](https://interledger.org/) payment connection via either STREAM or the Open Payments standard
+  - receive confirmation of the payment
+
+- The Web Monetization Provider is intended to be the entity that pays websites on behalf of the user. It's designed to be side-loaded via an extension, essentially allowing users to choose their provider by installing a different extension. It would only be responsible for offering the browser information about the rate, token, and provider address. Ideally, there would be a separate extension category for Web Monetization Providers.
```
[the remark this answers is not in the document]
Comment (AlexLakatos) [reply docs/web-monetization-api.md:22]: Even with OP, the provider would still "pay on behalf of the user", even if it's directing the funds movement from the user wallet. We ca change that to "`The Web Monetization Provider is intended to be the entity that facilitates payments to websites on behalf of the user.`"

In the current extension, the provider address would be the Coil BTP address. In an OP world, that would be the Create Incoming Payment endpoint I think.

File: docs/web-monetization-api.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+- The Web Monetization Provider is intended to be the entity that pays websites on behalf of the user. It's designed to be side-loaded via an extension, essentially allowing users to choose their provider by installing a different extension. It would only be responsible for offering the browser information about the rate, token, and provider address. Ideally, there would be a separate extension category for Web Monetization Providers. 
+
+Currently, the Agent and Provider are loaded via the same extension until the Web Monetization Agent is implemented in browsers.
+
+
+## Interfaces
+
+[`MonetizationEvent`](monetization-event.md)
+
+An event object that contains the amount, currency code, and receipt of a successful web monetization event.
```
[the remark this answers is not in the document]
Comment (AlexLakatos) [reply docs/web-monetization-api.md:31]: Receipts are still supported, so we need to document them. But we should add a Deprecated warning to them, and also document `incomingPayment`, maybe with an Experimental label to it.

File: docs/explainer.md [hunk 6]
```diff
@@ -203,182 +98,16 @@ identifiable as a payment account identifier.
...
-The Web Monetization provider (sender) interface leverages the
-[Payment Handler API](https://www.w3.org/TR/payment-handler/).
-
-More details are provided in the [Sending payments](sending.md) page.
+### Open Payments
+
+The Open Payments API provides a standard way for digital wallets to allow third-party applications to access and send funds based on the authorizations granted by the respective wallet owner.  
+Open Payments allows applications to access a digital wallet through one or more of its payment pointers. The payment pointer also serves as the entry point for the API.  
+
+For more information, see https://openpayments.dev.
```
Comment (sabineschaller) [inline docs/explainer.md:110]:
```suggestion
For more information, see https://openpayments.guide. 
```

**Review by sabineschaller — COMMENTED** [review]
I still need to finish reviewing but that is what I have so far.

File: docs/exclusive-content.md [hunk 7]
```diff
@@ -4,22 +4,25 @@ title: Exclusive Content
 sidebar_label: Exclusive content
 ---
 
+import Hidden from '@site/src/components/Hidden';
```
Comment (sabineschaller) [inline docs/exclusive-content.md:7]: Glitch examples didn't load for me on the preview

File: docs/exclusive-content.md [hunk 8]
```diff
@@ -198,17 +205,24 @@ If you see the source files instead of the example, click **View App** in the bo
...
   <iframe
-    src="https://glitch.com/embed/#!/embed/wm-exclusive-content-advanced?path=README.md&previewSize=100"
+    src="https://glitch.com/embed/#!/embed/wm2-exclusive-content-advanced?path=README.md&previewSize=100"
     title="wm-exclusive-content-advanced on Glitch"
     allow="geolocation; microphone; camera; midi; vr; encrypted-media"
     style={{ height: '100%', width: '100%', border: '0' }}>
   </iframe>
 </div>
 
+<Hidden>
```
Comment (sabineschaller) [inline docs/exclusive-content.md:215]: My Exclusive Content Generator is being removed 😢

Just kidding. I know it needs at least updating, if not a complete rethink.

File: docs/counter.md [hunk 9]
```diff
@@ -114,7 +110,7 @@ If you see the source files instead of the example, click **View App** in the bo
 
 <div class="glitch-embed-wrap" style={{ height: '420px', width: '100%' }}>
   <iframe
-    src="https://glitch.com/embed/#!/embed/wm-count-revenue?path=README.md&previewSize=100"
+    src="https://glitch.com/embed/#!/embed/wm2-count-revenue?path=README.md&previewSize=100"
```
Comment (sabineschaller) [inline docs/counter.md:113]: I think the "Thanks to you, I have made nothing (yet)" in the glitch a bit passive aggressive.

**Review by sabineschaller — COMMENTED** [review]
General comment:

We don't list the Puma Browser as Web Monetized Browser on the landing page anymore but it is still in the "Browser compatibility" table.

File: docs/web-monetization-api.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+The Web Monetization API provides a seamless payment experience for site owners and visitors. Unlike the W3C's [Payment Request API](https://developer.mozilla.org/en-US/docs/Web/API/Payment_Request_API), the Web Monetization API sends continuous micro-payments from the user agent to the website. Thus, visitors automatically send a small sum per time unit to any Web Monetized website they visit. The payment rate is predetermined by the user's Web Monetization agent and adapts over time.
+
+## Web Monetization Concepts and Usage
+
+The Web Monetization API allows web apps to accept continuous micro-payments from users. There are two components to this API:
+
+- The Web Monetization Agent is designed to be implemented in browsers and expose the Web Monetization API. It's responsible to:
+  - get the rate, token, and payment connection address from the Web Monetization Provider
+  - get the payment information from the `<link>` element
+  - create an [incoming payment](https://docs.openpayments.guide/reference/create-incoming-payment) via the [Open Payments API](https://docs.openpayments.guide)
```
[the remark this answers is not in the document]
Comment (mkurapov) [reply docs/web-monetization-api.md:20]: I think we should capitalize them, in line with other "interfaces" in the docs

[the remark this answers is not in the document]
Comment (mkurapov) [reply docs/web-monetization-api.md:20]: Would it make sense to specify on who's account the Incoming Payment is created on?

File: docs/web-monetization-api.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+[`MonetizationEvent`](monetization-event.md)
+
+An event object that contains all of the attributes of a successful web monetization event.
+
+## Basic example
+
+```html
+<link rel="monetization" href="https://example.com/pay" />
+```
```
Comment (mkurapov) [inline docs/web-monetization-api.md:39]: I think there should be more context here

File: docs/glossary.md [hunk 12]
```diff
@@ -4,32 +4,37 @@ title: Glossary
 sidebar_label: Glossary
 ---
 
+import Hidden from '@site/src/components/Hidden';
+
 # Monetized web page
```
Comment (mkurapov) [inline docs/glossary.md:9]: I wonder why this is a large title on the page (as compared to the other ones):
Screenshot 2023-05-23 at 13 39 52

File: docs/monetization-event-amount.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
+---
```
[the remark this answers is not in the document]
Comment (mkurapov) [reply docs/monetization-event-amount.md:1]: Do we want to keep the deprecated fields still? If we're going to be rolling out this new spec, do we care about the old interfaces? Those could just be on the/an old site

File: docs/monetization-event-incoming-payments.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+A string that represents a URL that can be used to verify payment at the [monetization receiver](glossary.md#web-monetization-receiver) via the [Open Payments API](https://docs.openpayments.guide/reference/get-incoming-payment) standard. 
+
+## Examples
+
+```javascript
+/** @type {MonetizationEvent} event */
+async function verifyPayment(event) {
+  // Legacy receivers don't support returning incoming payment URLs
+  if(!event.incomingPayment){
```
Comment (mkurapov) [inline docs/monetization-event-incoming-payments.md:28]:
```suggestion
  if (!event.incomingPayment){
```

File: docs/monetization-event-incoming-payments.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+  const response = await fetch('/verify', {
+    method: 'POST',
+    credentials: 'same-origin',
+    mode: 'same-origin',
+    cache: 'no-cache',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body);
+  });
```
[the remark this answers is not in the document]
Comment (mkurapov) [reply docs/monetization-event-incoming-payments.md:45]: I was going to comment the same thing when I saw this on the site

File: docs/monetization-event-incoming-payments.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+
+  if (response.ok) {
+    // The incoming payment was fetched successfully
+    const { receivedAmount } = JSON.parse(response.json())
+    const { amount, assetCode, assetScale } = receivedAmount;
```
[the remark this answers is not in the document]
Comment (mkurapov) [reply docs/monetization-event-incoming-payments.md:50]: Why IF? `receivedAmount` is what we'd get in Open Payments, right

File: sidebars.json [hunk 17]
```diff
@@ -1,9 +1,58 @@
...
+          "link": {
+            "type": "doc",
+            "id": "permission-policy"
+          },
+          "items": ["permission-policy-monetization"]
+        }
+      ]
+    },
+    {
+      "Concepts": ["explainer", "receiving", "sending"],
```
Comment (mkurapov) [inline sidebars.json:47]: _Maybe_ "Concepts" should go before "Interfaces"? I think there needs to be an explainer/getting started section should go before the devvy part

Comment (mkurapov) [reply sidebars.json:47]: Or even a separate Tab altogether for "API"

File: docs/glossary.md (hunk 12 above)
Comment (mkurapov) [reply docs/glossary.md:9]: Maybe we can just convert all of these to use ## (header 2) size, and just add `# Glossary` at the beginning

File: docs/exclusive-content.md [hunk 18]
```diff
@@ -47,18 +51,18 @@ Web Monetization makes providing exclusive content easy! This is a very simple e
...
-the state goes from `pending` to `started`.
+Next, we add an event listener to the `link` element. The
+`monetization` event is emitted when Web Monetization initializes.
 
 ```js
-document.monetization.addEventListener('monetizationstart', () => {
+link.addEventListener('monetization', ev => {
 ```
 
 Finally, we select our exclusive content element and make it visible. We defined a CSS class that made it hidden, so removing that class will make it visible. If you want to do something else when Web Monetization starts, you can replace this line. You can trigger any JavaScript, so the sky's the limit.
```
Comment (mkurapov) [inline docs/exclusive-content.md:68]: We should fix the formatting on L67 to be:
```js
document.getElementById('exclusive').classList.remove('hidden')
```

**Review by mkurapov — COMMENTED** [review]
First pass. I also had some notes for the Concepts section (https://webmonetization-preview.netlify.app/docs/explainer), but I can do that separately since it wasn't in this diff

File: docs/web-monetization-api.md [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+  <Hidden>- create an [Interledger](https://interledger.org/) payment connection via either the [STREAM](https://interledger.org/rfcs/0029-stream/) protocol or the [Open Payments](https://docs.openpayments.guide/) standard</Hidden>
+  - verify payment via the [outgoing payment](https://docs.openpayments.guide/reference/create-outgoing-payment) endpoint on the web site owner's payment pointer via the [Open Payments API](https://docs.openpayments.guide)
+
+- The Web Monetization Provider is intended to be the entity that facilitates payments to websites on behalf of the user. It's designed to be side-loaded via an extension, essentially allowing users to choose their provider by installing a different extension. It would only be responsible for offering the browser information about the rate, token, and provider address. Ideally, there would be a separate extension category for Web Monetization Providers.  A website visitor can also potentially be their own Web Monetization Provider through the use of a digital wallet that implements the [Open Payments API](https://openpayments.guide) standard.
```
[the remark this answers is not in the document]
Comment (mkurapov) [reply docs/web-monetization-api.md:24]: I also don't think we need the "Ideally, there would be a separate extension category for Web Monetization Providers."  statement

File: docs/monetization-event-receipt.md [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+A base64-encoded [STREAM receipt](https://github.com/interledger/rfcs/blob/master/0039-stream-receipts/0039-stream-receipts.md). Returns `null` if receipts are not enabled on the payment pointer. The receipt has the total running amount streamed in the current session.
+
+## Examples
+
+```html
+<link rel="monetization" href="https://example.com/pay" />
+<script>
+  const link = document.querySelector(“link[rel=‘monetization’]“);
+  link.addEventListener(“monetization”, (event) => {
+  console.log(event.receipt)})
```
[the remark this answers is not in the document]
Comment (mkurapov) [reply docs/monetization-event-receipt.md:32]: Related to another comment, I kind of think we shouldn't including anything deprecated on this site (including receipts)

**Comment by mkurapov** [comment]
Related to the comment about deprecated fields: maybe we can have a v1/v2 API toggle (like https://docusaurus.io/ versioning)

**Comment by sublimator** [comment]
Where were you last year, Max?
Version toggles on docs!
What a crazy concept!

File: docs/monetization-event-incoming-payments.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (sublimator) [reply docs/monetization-event-incoming-payments.md:45]: I guess at the time they assumed only the backend could have access to the incomingPayment to verify it ? Which seems a reasonable assumption ?

File: docs/web-monetization-api.md [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+
+## Web Monetization Concepts and Usage
+
+The Web Monetization API allows web apps to accept continuous micro-payments from users. There are two components to this API:
+
+- The Web Monetization Agent is designed to be implemented in browsers and expose the Web Monetization API. It's responsible to:
+  - get the rate, token, and payment connection address from the Web Monetization Provider
+  - get the payment information from the `<link>` element
+  - create an [incoming payment](https://docs.openpayments.guide/reference/create-incoming-payment) via the [Open Payments API](https://docs.openpayments.guide)
+  <Hidden>- create an [Interledger](https://interledger.org/) payment connection via either the [STREAM](https://interledger.org/rfcs/0029-stream/) protocol or the [Open Payments](https://docs.openpayments.guide/) standard</Hidden>
```
[the remark this answers is not in the document]
Comment (hajjimo) [reply docs/web-monetization-api.md:21]: The thinking here was to remove ILP/STREAM mentions and steer toward OP.

**Pull request merged**